base/install.sh

```diff
@@ -33,29 +46,76 @@ chgrp ssh_keys /etc/ssh/ssh_host_rsa_key
 chgrp ssh_keys /etc/ssh/ssh_host_ecdsa_key
 chgrp ssh_keys /etc/ssh/ssh_host_ed25519_key
 
+#------------------------
+# Setup LDAP and SSSD
+#------------------------
+log_info "Configuring LDAP and SSSD"
+authconfig --enableldap \
+  --enableldapauth \
+  --ldapserver=ldap://ldap:389 \
+  --ldapbasedn="dc=example,dc=org" \
+  --enablerfc2307bis \
+  --enablesssd \
+  --enablesssdauth \
+  --kickstart \
+  --nostart \
+  --update
+
+cat > /etc/openldap/ldap.conf <<EOF
+TLS_CACERTDIR /etc/openldap/cacerts
+TLS_REQCERT never
+SASL_NOCANON	on
+URI ldaps://ldap:636
+BASE dc=example,dc=org
+EOF
+
+cat > /etc/sssd/sssd.conf <<EOF
+[domain/default]
+debug_level = 3
+autofs_provider = ldap
+ldap_schema = rfc2307bis
+ldap_group_member = member
+ldap_search_base = dc=example,dc=org
+id_provider = ldap
+auth_provider = ldap
+chpass_provider = ldap
+ldap_uri = ldaps://ldap:636
+cache_credentials = True
+ldap_tls_reqcert = never
+ldap_default_bind_dn = cn=admin,dc=example,dc=org
+ldap_default_authtok = admin
+
+[sssd]
+debug_level = 3
+services = nss, pam
+domains = default
+
+[nss]
+debug_level = 3
+homedir_substring = /home
+
+[pam]
+debug_level = 3
+EOF
+
 #------------------------
 # Setup user accounts
 #------------------------
 
-idnumber=1000
+idnumber=1001
 for uid in hpcadmin $USERS
 do
-    log_info "Creating $uid user account with uidnumber $idnumber.."
-    passvar="PASSWD_$uid"
-    passwd=${!passvar:-ilovelinux}
-    groupadd --gid $idnumber $uid
-    useradd  --gid $idnumber --uid $idnumber $uid
-    echo -n $passwd  | passwd --stdin $uid
-	install -d -o $uid -g $uid -m 0700 /home/$uid/.ssh
-    sudo -u $uid ssh-keygen -b 2048 -t rsa -f /home/$uid/.ssh/id_rsa -q -N ""
-    install -o $uid -g $uid -m 0600 /home/$uid/.ssh/id_rsa.pub /home/$uid/.ssh/authorized_keys
-	sudo -u $uid tee /home/$uid/.ssh/config <<EOF
+    log_info "Bootstrapping $uid user account.."
+    install -d -o $idnumber -g $idnumber -m 0700 /home/$uid/.ssh
+    ssh-keygen -b 2048 -t rsa -f /home/$uid/.ssh/id_rsa -q -N ""
+    install -o $idnumber -g $idnumber -m 0600 /home/$uid/.ssh/id_rsa.pub /home/$uid/.ssh/authorized_keys
```

Comment:
Now that we're not using `useradd` we don't get the skel files copied over for bash. Can we add something like:
```
cp /etc/skel/.bash* /home/$uid`
chown $uidnumber:$idnumber /home/$uid/.bash*
```

Reply:
Added this change.

Reply:
OK think we're super close. Still missing some chmods on the `.bash*` files:

```
[hpcadmin@frontend ~]$ ls -alh
total 28K
drwx------ 3 hpcadmin hpcadmin 4.0K Jul  6 17:48 .
drwxr-xr-x 7 root     root     4.0K Jul  6 17:42 ..
-rw-r--r-- 1 root     root       18 Jul  6 17:30 .bash_logout
-rw-r--r-- 1 root     root      193 Jul  6 17:30 .bash_profile
-rw-r--r-- 1 root     root      231 Jul  6 17:30 .bashrc
drwx------ 2 hpcadmin hpcadmin 4.0K Jul  6 17:48 .ssh
-rw------- 1 hpcadmin hpcadmin  610 Jul  6 17:48 .viminfo
```

Reply:
I believe we're missing the `-R` here:
```
    chown $idnumber:$idnumber /home/$uid
```

Reply:
Yep, fixed and removed one redundant chown.